### PR TITLE
Add "ray status" debug tool for autoscaler.

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This adds a "ray status" cluster debug tool for the autoscaler. It can be used to inspect the current autoscaling state instead of trying to read cluster logs.

Also, some minor improvements to autoscaler logging.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/scripts/scripts.py [hunk 1]
```diff
@@ -1094,6 +1095,21 @@ def memory(address):
...
+    required=False,
+    type=str,
+    help="Override the address to connect to.")
+def status(address):
+    """Print cluster status, including autoscaling info."""
+    if not address:
+        address = services.find_redis_address_or_die()
+    logger.info("Connecting to Ray instance at {}.".format(address))
+    ray.init(address=address)
+    print(debug_status())
```
Comment (richardliaw) [inline python/ray/scripts/scripts.py:1110]: Hey, wouldn't this be more useful if it were used like:

`ray status [cluster_yaml]`, similar to `ray dashboard cluster.yaml`?

**Comment by richardliaw** [comment]
Looks very useful, can you post a print output?

File: python/ray/scripts/scripts.py (hunk 1 above)
Comment (ericl) [reply python/ray/scripts/scripts.py:1110]: There are a number of ray commands that are intended to run on the currently active cluster such as ray memory, etc. This is consistent with those.

**Comment by ericl** [comment]
The output looks like this:
```
Cluster status: 0/0 target nodes (0 pending)
 - MostDelayedHeartbeats: {}
 - NodeIdleSeconds: Min=-1 Mean=-1 Max=-1
 - NumNodesConnected: 0
 - NumNodesUsed: 0.0
 - ResourceUsage: 
 - TimeSinceLastHeartbeat: Min=-1 Mean=-1 Max=-1
```

File: python/ray/autoscaler/commands.py [hunk 2]
```diff
@@ -80,8 +109,11 @@ def teardown_cluster(config_file, yes, workers_only, override_cluster_name,
     confirm("This will destroy your cluster", yes)
 
     if not workers_only:
-        exec_cluster(config_file, "ray stop", False, False, False, False,
-                     False, override_cluster_name, None, False)
+        try:
+            exec_cluster(config_file, "ray stop", False, False, False, False,
+                         False, override_cluster_name, None, False)
+        except Exception:
+            logger.exception("Ignoring error attempting a clean shutdown.")
```
Comment (ericl) [inline python/ray/autoscaler/commands.py:116]: @ijrsvt otherwise teardown fails if ray is misconfigured

**Review by wuisawesome — APPROVED** [review]
lgtm

**Pull request merged**